### DISCUSSION TITLE
Living with Broken Software - How to Plot Chia without going insane

### DISCUSSION THREAD

**Discussion by room101-dev** [Chia-Network/chia-blockchain#5568] [General]
1.) Fall back to the last working plotter chia-pos 1.3, all the software since then is broken

2.) Use linux, and cmd-line only

3.) Don't use any gui tools

4.) don't use any plot managers like 'plotman' they'll kill all your hard work, and reset you to scratch

5.) develop monitors for all NVME iostat, nvme, htop are your friends

6.) never try to run more than 4x plots in parallel on one NVME drive, the controller can't handle it

7.) Watch for the infamous 'mis-match' error in the chai-log for each 'chia create plot' instance, once you get that message, it means that NVME has been destroyed

8.) Watch 'sudo nvme smart-log /dev/nvmeNNN', watch TEMP of controller and 'controller busy count' if either go astro, then kill that plot process before it kills your NVME

9.) Frequently run "sudo fstrim -v /dev/nvmeNNN' on the drive to trim free memory for the drive. When formatting use ZFS, and 'discard' to mark the disk, but you still need to manual run fstrim every 4 plots on every NVME otherwise the plotting will slow down to zero. The NVME drives don't have self cleaning, and the brutal abuse by 'chia-pos' requires frequent trimming.

I see now that chia-pos is saying they have a new 'killer version' ( 14), since about 1.0 of chia-plot, 'making it faster' has really meant killing NVME's more often, the plot speed is always going to be 6-8 hours, its always best NOT TO KILL YOUR HW. CHIA-DEV doesn't care about you, they only care about the CHIA IPO.

The GUI is unstable, the blockchain code is now terminal, nobody on earth can get chia-net to sync, and if you do, its doesn't stay sync'd for long.

The only thing useful is to use old pre 1.3 plotting software and make plots and then mine on HPOOL.com and make some money today.

Once chia gets the blockchain stable and sync'd ( this could take months, if ever ), then we will see. My fear for now is that the newer plotter software is designed to make it impossible to earn money on HPOOL, so there is a war, don't get involved, hold back to a working version, where the plots are supported in both worlds.

DEV is saying that when their plotting protocol is done, we must destroy all our plots and start over, the chinese now own chia by 70%, they'll not tolerate this bullshit coming from dev. The CHINESE IOU's made the price go to $1600usd, and keep it there, once the chia-dev team trys to screw china out  of its hard-work, then chia will implode.

IMHO the future is this, people  (US USERS) will clone chia fix the bugs, as its clear this team has no software management experience at delivering a software product to a commercial market. Sure they're all great influencers and marketing guys, but they have never developed software in a real software company.

Chia is a good idea, its too bad that a weak management team destroyed it. It's not the first proof-of-space coin, there will be others, we all can see the problems, so now lets fix them.

Sure BRAM created bit-torrent yaba-yaba, but it took 5+ years to get it rock solid, and he was probably long gone the people that made it rock solid were more than likely dev's in the wild improving open-source.

...

I cannot stress enough, why plot chia if you can't earn XCH? To do this you must use HPOOL.COM, they have a plotter, and a miner, I only use their miner, its rock solid, and they do updates weekly; I don't know about their plotter I don't use it, I don't have to give them my keys, I just sign a their generated public-key, and it proves I own my plots. Right now you can earn a chia every two months, with every 200 plots. If you do as your told and 'solo mine' you will earn ZERO, and then be told to delete your plots once the pool is blessed by god-chia. They're even saying your must POS to join their pool, this of course is all 100% bullshit.

HPOOL.COM is fair & honest company; the guy is western educated and speaks fluent english; its just a normal pool just like ethermine.org, or superpool.io; These guys have years of experience, the chia-team don't know shit about mining-pools, they'll never deliver anything but bugs.

Lastly the question why have for months the talking heads on youtube dissed HPOOL.COM, these are all paid influencers, by you know who. Why the lies? About private-keys, about stealing, about not trusting china? Why? 

Why do they do nothing about destroying our NVME drives, why do they not care?

Why is chia-dev sw management so blatantly incompetent?

Why is every reported problem on github 'bug-fix' called to be a 'ram' problem, run memtest; Is github ran by a sentiment bot for chia? sub 100 iq? or Is chia a tier 3 operation?

**Comment by SackofFluor** [comment]
i really wanted to lend some credibility to your rambling (again... lol)
 
BUT THEN:
"HPOOL.COM is fair & honest company; the guy is western educated and speaks fluent english; its just a normal pool just like ethermine.org, or superpool.io; These guys have years of experience, the chia-team don't know shit about mining-pools, they'll never deliver anything but bugs."

Yes, its SOOOO fair and honest that the homepage doesn't even have an English version.....
So when I looked for a guide I found a Russian guy using the Chinese version to help people understand it....

you complain that the dev team may be SOOOO incompetent, but you guys (and lets face it, you are affiliated with hpool) are incapable of getting an English version, an English homepage or at least a proper English how-to?

You constantly ramble about how bad CHIA is... dude then just stop and agitate yourself about this coin....

**Reply by climba03003** [reply]
He first saying `Chia POS is top secret and not published publicly`, then we post the Github link to show it is not the truth.
Then, he mention about `ASIC Plotting`, it is not really a profitable for making an `ASIC Plotting`.

Here the third article is just blaming the devs but not to give a hand on sending PR and improve the codebase.

**Comment by grasuu** [comment]
I think the only way to stay sane is to understand that plotting speed doesn't matter. 

What limits you is the cost of hdds, the space and infrastructure you need to run them in and energy costs.
You can run a datacenter drive 10 to 15 years. If you fill it with plots in 5 or 10 or 25 days really doesn't matter in the end (you run it for several 1000 days, right?)

**Reply by FSys77** [reply]
Do you understand the concept of "difficulty"? Obviously not. The game is: Bring as much plots as possible AS SOON AS POSSIBLE, before difficulty sky rockets.

**Reply by grasuu** [reply]
Sure, it's easy enough.

And -- honestly -- we are by far over that point already. I could explain more, but do the math yourself, it's quite easy.

**Comment by grasuu** [comment]
> 9.) Frequently run "sudo fstrim -v /dev/nvmeNNN' on the drive to trim free memory for the drive. When formatting use ZFS, and 'discard' to mark the disk, but you still need to manual run fstrim every 4 plots on every NVME otherwise the plotting will slow down to zero. The NVME drives don't have self cleaning, and the brutal abuse by 'chia-pos' requires frequent trimming.

Sorry, I'm afraid you got a lot of things wrong here. ZFS is an excellent file system, but by far the wrong choice for a scratch disk. It's just the wrong tool for the task. It's a COW file system, and well known for becomming really slow if utilized over 80%.

**Comment by mariano54** [comment]
We are working on fixing all of these issues (and getting pools out really quick). I think given how fast we have grown, and only launched less than 2 months ago, it is working pretty well.

That said, we do understand your pain and really want to get all of these bugs fixed. No new innovative technology is perfectly stable from the start... So please have some patience! We are also scaling the team and external contributors so more changes and improvements will be coming soon

**Reply by FSys77** [reply]
"No new innovative technology is perfectly stable from the start". Blockchain is not exactly new technology. And test net was online a long time ago. Still you stuff is buggy as hell.

**Reply by chiabug** [reply]
I have been developing software for a VERY long time.  I have to agree that even though testnet was online for a long time Chia was still not ready for prime time.  There is far more to production ready software than just whether it will run on the latest version of hardware.
